realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -407,7 +411,7 @@ public boolean hasPrimaryKey() {
      */
     public Set<String> getFieldNames() {
         int columnCount = (int) table.getColumnCount();
-        Set<String> columnNames = new TreeSet<String>();
+        Set<String> columnNames = new LinkedHashSet<>(columnCount);
```

Comment:
Insertion order, so people get the order of fields in core when they iterator over the set.